wandb/sdk/wandb_run.py

```diff
@@ -935,15 +933,14 @@ def tags(self, tags: Sequence) -> None:
     @_run_decorator._attach
     def id(self) -> str:
         """Identifier for this run."""
-        if TYPE_CHECKING:
-            assert self._settings.run_id is not None
-        return self._settings.run_id
+        assert self._settings.run_id is not None
+        return self._settings.run_id or ""
```

Comment:
The `or ""` is redundant because the assert ensures that `run_id` is not `None`.